GOLDTracker/src/components/SearchComponent.js

```diff
@@ -42,95 +111,172 @@ const SearchComponent = ({ search, setSearch, setIsSearching, major }) => {
+        toValue: 0,
+        duration: 0,
+        useNativeDriver: false,
+      }).start();
     }
+    Animated.timing(animatedDeptHeight, {
+      toValue: isDeptDropdownVisible ? 0 : 300,
+      duration: 300,
+      easing: Easing.inOut(Easing.ease),
+      useNativeDriver: false,
+    }).start();
+  };
 
-    try {
-      const quarter = selectedQuarter; 
-      // Prioritize deptCode; fallback to text input
-      // If major is not an empty string, use the major as deptCode
+  const toggleQuarterDropdown = () => {
+    if (!isQuarterDropdownVisible) {
+      setIsSearching(true);
+    } else if (!isDeptDropdownVisible) {
+      setIsSearching(false);
+    }
+    setIsQuarterDropdownVisible((prev) => !prev);
+    if (!isQuarterDropdownVisible) {
+      setIsDeptDropdownVisible(false);
+      Animated.timing(animatedDeptHeight, {
+        toValue: 0,
+        duration: 0,
+        useNativeDriver: false,
+      }).start();
+    }
+    Animated.timing(animatedQuarterHeight, {
+      toValue: isQuarterDropdownVisible ? 0 : 300,
+      duration: 300,
+      easing: Easing.inOut(Easing.ease),
+      useNativeDriver: false,
+    }).start();
+  };
+
+  const handleSelect = (type, item) => {
+    if (type === "department") {
+      if (item.code === "Your Major") {
+        setSelectedDept(major);
+        handleSearchSubmit(major);
+      } else {
+        setSelectedDept(item.code);
+        handleSearchSubmit(item.code);
+      }
+      toggleDeptDropdown();
+    } else if (type === "quarter") {
+      setSelectedQuarter(item.code);
+      toggleQuarterDropdown();
+    }
+  };
+
+  const closeDropdowns = () => {
+    if (isDeptDropdownVisible || isQuarterDropdownVisible) {
+      setIsSearching(false);
+    }
+    if (isDeptDropdownVisible) {
+      setIsDeptDropdownVisible(false);
+      Animated.timing(animatedDeptHeight, {
+        toValue: 0,
+        duration: 300,
+        useNativeDriver: false,
+      }).start();
+    }
+    if (isQuarterDropdownVisible) {
+      setIsQuarterDropdownVisible(false);
+      Animated.timing(animatedQuarterHeight, {
+        toValue: 0,
+        duration: 300,
+        useNativeDriver: false,
+      }).start();
+    }
+  };
 
+  useImperativeHandle(categorySearchRef, () => ({
+    closeDropdowns,
+  }));
+
+  const handleSearchSubmit = async (deptCode = null) => {
+    setIsSearching(true);
+    setIsLoading(true);
+
+    let searchTerm = deptCode || search.trim();
+
+    try {
+      const quarter = selectedQuarter;
 
       if (!searchTerm) {
         setErrorMessage("Please enter a search term or select a department.");
+        setIsSearching(false); // Stop searching
         return;
       }
 
-      console.log("department code in search is: ", deptCode)
-      console.log("search.trim in search is: ", search.trim())
-      apiUrl = ""
-      if (deptCode){
-        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)
-
+      const apiUrl = /^[A-Z]{2,}\s[\dA-Z]+$/.test(searchTerm)
```

Comment:
For subject area search and department search, we should use two different parameters in API. You can see the difference in current main branch. 

      if (deptCode){
        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)

        ? `${API_URL}?quarter=${quarter}&courseId=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true`
        : `${API_URL}?quarter=${quarter}&deptCode=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true&pageNumber=1&pageSize=30`;
      }
      else{
        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)
        ? `${API_URL}?quarter=${quarter}&courseId=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true`
        : `${API_URL}?quarter=${quarter}&subjectCode=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true&pageNumber=1&pageSize=30`;
      }